components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/http/management/HTTPEndpointManager.java

```diff
@@ -426,20 +435,37 @@ protected Pattern compilePattern(String dispatchPattern) {
         }
     }
 
-    public InternalAPIDispatcher getInternalAPIDispatcher() {
-        return internalAPIDispatcher;
+    public InternalAPIDispatcher getInternalHttpAPIDispatcher() {
+        return internalHttpAPIDispatcher;
+    }
+
+    public InternalAPIDispatcher getInternalHttpsAPIDispatcher() {
+        return internalHttpsAPIDispatcher;
     }
 
     public int getInternalInboundPort() {
         return internalInboundPort;
     }
 
+    public int getInternalHTTPSInboundPort() {
+        return internalInboundHttpsPort;
+    }
+
     /**
      * Checks whether at least one internal API is enabled.
      *
      * @return - whether any internal API is enabled.
      */
-    public boolean isAnyInternalApiEnabled() {
-        return internalApiEnabled;
+    public boolean isAnyInternalHttpApiEnabled() {
```

Comment:
Why "Any" part is introduced to the variable name? Is there multiple Internal APIs?

Reply:
yes there are